ui/app/src/components/PublishingDashboard/PublishingDashboard.tsx

```diff
@@ -0,0 +1,254 @@
+        });
+      }
+    );
+  };
+
+  const bulkPublishConfirmation = () => {
+    dispatch(
+      showConfirmDialog({
+        body: `${formatMessage(messages.publishStudioWarning)} ${formatMessage(messages.publishStudioNote)}`,
+        onCancel: batchActions([closeConfirmDialog(), dispatchDOMEvent({ id: idCancel })]),
+        onOk: batchActions([closeConfirmDialog(), dispatchDOMEvent({ id: idSuccess })])
+      })
+    );
+
+    const successCallback = () => {
+      bulkPublish();
+      document.removeEventListener(idSuccess, successCallback, false);
+      document.removeEventListener(idCancel, cancelCallback, false);
+    };
+    const cancelCallback = () => {
+      document.removeEventListener(idCancel, cancelCallback, false);
+      document.removeEventListener(idSuccess, successCallback, false);
+    };
+
+    document.addEventListener(idSuccess, successCallback, false);
+    document.addEventListener(idCancel, cancelCallback, false);
+  };
+
+  const publishBy = () => {
+    const { commitIds, environment, comment } = publishGitFormData;
+    const ids = commitIds.replace(/\s/g, '').split(',');
+
+    publishByCommits(site, ids, environment, comment).subscribe(
+      () => {
+        dispatch(
+          showSystemNotification({
+            message: formatMessage(messages.publishSuccess)
+          })
+        );
+        setPublishGitFormData(initialPublishGitFormData);
+      },
+      ({ response }) => {
+        dispatch(
+          showSystemNotification({
+            message: response.message,
+            options: { variant: 'error' }
+          })
+        );
+      }
+    );
+  };
+
+  const onCancelPublishOnDemand = () => {
+    setPublishOnDemandMode(null);
+    setPublishStudioFormData(initialPublishStudioFormData);
+    setPublishGitFormData(initialPublishGitFormData);
+  };
+
+  return (
+    <section className={classes.root}>
+      <Grid container spacing={3}>
+        <Grid item xs={12}>
+          <PublishingStatusWidget state={state} onStartStop={onStartStop} onRefresh={onRefresh} onUnlock={onUnlock} />
+        </Grid>
+        <Grid item xs={12}>
+          <PublishOnDemandWidget
+            mode={publishOnDemandMode}
+            setMode={setPublishOnDemandMode}
+            formData={publishOnDemandMode === 'studio' ? publishStudioFormData : publishGitFormData}
+            setFormData={publishOnDemandMode === 'studio' ? setPublishStudioFormData : setPublishGitFormData}
+            formValid={publishOnDemandMode === 'studio' ? publishStudioFormValid : publishGitFormValid}
+            publishingTargets={publishingTargets}
+            publishingTargetsError={publishingTargetsError}
+            onPublish={publishOnDemandMode === 'studio' ? bulkPublishConfirmation : publishBy}
+            note={formatMessage(
+              publishOnDemandMode === 'studio' ? messages.publishStudioNote : messages.publishGitNote
+            )}
+            onCancel={onCancelPublishOnDemand}
+          />
```

Comment:
Let's make the widget(s) the containers. Otherwise the dashboard will be too much. The dashboard should end up lighter, mostly controlling which widgets to load, the layout and —if any— common resources, but logic of each widget goes to the widget.